operator/pkg/gateway-api/cell.go

```diff
@@ -221,58 +226,131 @@ func checkCRD(ctx context.Context, clientset k8sClient.Clientset, gvk schema.Gro
 	return nil
 }
 
-func checkRequiredCRDs(ctx context.Context, clientset k8sClient.Clientset) error {
+// checkCRDs checks if required and optional CRDs are present in the cluster,
+// returns an error if the required CRDs are not installed, and returns the
+// schema.GroupVersionKind of any optional CRDs that are installed.
+func checkCRDs(ctx context.Context, clientset k8sClient.Clientset, logger *slog.Logger) ([]schema.GroupVersionKind, error) {
 	var res error
-	for _, gvk := range requiredGVK {
+	var presentGVKs []schema.GroupVersionKind
+
+	for _, gvk := range requiredGVKs {
 		if err := checkCRD(ctx, clientset, gvk); err != nil {
 			res = errors.Join(res, err)
 		}
 	}
-	return res
+
+	for _, optionalGVK := range optionalGVKs {
+		if strings.HasSuffix(optionalGVK.Kind, "lists") {
+			continue
+		}
+		if err := checkCRD(ctx, clientset, optionalGVK); err == nil {
```

Comment:
nit: i would invert the check and make an early `continue` to avoid the need for the `else` case and the nesting.

```
if err := checkCRD(ctx, clientset, optionalGVK); err != nil {
  logger.Debug("CRD is not present, will not handle it", "optionalGVK", optionalGVK.String())
  continue
}

// present handling...
```

Reply:
of course, that's way more idiomatic, thanks!